transport/policy.go

```diff
@@ -256,8 +210,30 @@ func (p *NetworkTopologyTokenAwarePolicy) Update(t *topology) {
 			}
 		}
 	}
+	return &NetworkTopologyTokenAwarePolicy{
+		WrapperPolicy: p.WrapperPolicy.New(t, stg).(WrapperPolicy),
+		PreparedNodes: preparedNodes,
+	}
 }
 
-func (p *NetworkTopologyTokenAwarePolicy) Clone() HostSelectionPolicy {
-	return p // TODO: implement valid Clone method.
+func NewNetworkTopologyTokenAwarePolicy(wp WrapperPolicy) *NetworkTopologyTokenAwarePolicy {
+	return &NetworkTopologyTokenAwarePolicy{
+		WrapperPolicy: wp,
+	}
+}
+
+func (p NetworkTopologyTokenAwarePolicy) Iter(qi QueryInfo, idx int) *Node {
+	start, end := 0, len(p.PreparedNodes)
```

Comment:
Shouldn't it be 
```go
start, end := 0, len(p.PreparedNodes)-1
```
?

Reply:
Then I believe you can delete this
```go
if start == len(p.PreparedNodes) {
 	start = 0
}
```

Reply:
Answers:
- no
- no

Reply:
Tokens: [1, 3, 5, 7]
Query: 8
Got: 7
Wanted: 1